engine/src/conversion/analysis/fun/mod.rs

```diff
@@ -1610,27 +1694,40 @@ impl<'a> FnAnalyzer<'a> {
     /// If a type has explicit constructors, bindgen will generate corresponding
     /// constructor functions, which we'll have already converted to make_unique methods.
     /// C++ mandates the synthesis of certain implicit constructors, to which we
-    /// need ro create bindings too. We do that here.
+    /// need to create bindings too. We do that here.
     /// It is tempting to make this a separate analysis phase, to be run later than
     /// the function analysis; but that would make the code much more complex as it
     /// would need to output a `FnAnalysisBody`. By running it as part of this phase
     /// we can simply generate the sort of thing bindgen generates, then ask
     /// the existing code in this phase to figure out what to do with it.
-    fn add_missing_constructors(&mut self, apis: &mut Vec<Api<FnPrePhase>>) {
-        if self.config.exclude_impls {
-            return;
-        }
-        let implicit_constructors_needed = find_missing_constructors(apis);
-        for (self_ty, implicit_constructors_needed) in implicit_constructors_needed {
+    ///
+    /// Also fills out the [`PodAndConstructorAnalysis::constructors`] fields with information useful
+    /// for further analysis phases.
+    fn add_constructors_present(&mut self, apis: &mut Vec<Api<FnPrePhase>>) {
+        for (self_ty, items_found) in find_constructors_present(apis).iter() {
+            if let Some(Api::Struct {
+                analysis: PodAndConstructorAnalysis { constructors, .. },
+                ..
+            }) = apis.iter_mut().find(|api| api.name() == self_ty)
```

Comment:
I'm concerned this may O(n^2) over the number of APIs. I think I've avoided that elsewhere so far...

Reply:
Looks like you fixed this one in #889, LGTM.

Reply:
Never mind, you just cut the square in half, but the triangle is still O(n^2). This one is pretty easy to move into `find_constructors_present` instead, done.